tendrl/monitoring_integration/sync/__init__.py

```diff
@@ -11,6 +16,28 @@ def __init__(self):
         self._complete = gevent.event.Event()
 
     def _run(self):
-        while not self._complete.is_set():
-            webhook_receiver.start_server()
+        interval = etcd_utils.read("_NS/gluster/config/data/sync_interval")
+
+        try:
 
+            interval = float(interval.value)
+
+        except ValueError as ex:
+            logger.log("error", NS.get("publisher_id", None),
+                       {'message': str(ex)})
+            raise ex
+
+        while not self._complete.is_set():
+            try:
+                gevent.sleep(interval)
+                plugin_obj = GraphitePlugin()
```

Comment:
Do you really need to create instance of GraphitePlugin every iteration? Move it outside the while loop, and in the GraphitePlugin class handle any socket read/write errors